navigator/backend/src/main/scala/com/digitalasset/navigator/store/platform/PlatformStore.scala

```diff
@@ -159,37 +162,72 @@ class PlatformStore(
       stash()
   }
 
+  @SuppressWarnings(
+    Array(
+      "org.wartremover.warts.JavaSerializable",
+      "org.wartremover.warts.Product",
+      "org.wartremover.warts.Serializable",
+    )
+  ) // for the andThen below
   def connected(state: StateConnected): Receive = {
-    case UpdateParties =>
-      state.ledgerClient.partyManagementClient
-        .listKnownParties()
-        .map(UpdatedParties(_))
-        .pipeTo(self)
+    case UpdateUsersOrParties =>
+      state.ledgerClient.versionClient
+        .getApiFeatures(state.ledgerClient.ledgerId)
+        .filter(features => // if we have user management (and it's not disabled)....
+          enableUserManagement && features.contains(Feature.UserManagement)
+        )
+        .andThen {
+          case Success(_) => // .. then only list users on the login screen
+            state.ledgerClient.userManagementClient
+              .listUsers() // don't pass token here -- it's already set on startup (LedgerClientConfiguration)
+              .map(UpdatedUsers(_))
+              .pipeTo(self)
+          case Failure(_) => // ... else fallback to parties
+            state.ledgerClient.partyManagementClient
+              .listKnownParties()
+              .map(UpdatedParties(_))
+              .pipeTo(self)
+        }
+
       ()
+
+    case UpdatedUsers(users) =>
+      // Note: you cannot log in as a user without a primary party
```

Comment:
If I understand correctly if this invariant is not respected this will fail silently, right? Should we log something if the primary party is absent?

Reply:
I'll add some logging.

Reply:
FTR, I'm not convinced the complexity required to log this is worth it, but I've added it as follows:
```
      users.filter(_.primaryParty.isEmpty) match {
        case usersWithoutPrimaryParty if !usersWithoutPrimaryParty.isEmpty =>
          log.warning(
            s"Users without primary party (counted ${usersWithoutPrimaryParty.length})" +
              s" cannot be used for login ${usersWithoutPrimaryParty.take(10).map(_.id.toString).mkString("(e.g., ", ", ", ")")}."
          )
      }
```